internal/gps/vcs_source.go

```diff
@@ -304,16 +305,20 @@ func (s *gopkginSource) listVersions(ctx context.Context) ([]PairedVersion, erro
 		case branchVersion:
 			// The semver lib isn't exactly the same as gopkg.in's logic, but
 			// it's close enough that it's probably fine to use. We can be more
-			// exact if real problems crop up. The most obvious vector for
-			// problems is that we totally ignore the "unstable" designation
-			// right now.
+			// exact if real problems crop up.
 			sv, err := semver.NewVersion(tv.name)
 			if err != nil || sv.Major() != s.major {
 				// not a semver-shaped branch name at all, or not the same major
 				// version as specified in the import path constraint
 				continue
 			}
 
+			// Gopkg.in has a special "-unstable" suffix which we need to handle
+			// separately.
+			if s.unstable && !strings.HasSuffix(tv.name, gopkgUnstableSuffix) {
```

Comment:
Looking at [these docs](https://github.com/niemeyer/gopkg/blob/master/version.go#L56-L57), I think the expected behavior for gopkg.in is that the set of versions for stable vs. unstable are disjoint. That would suggest that we probably need to check for the `-unstable` suffix in both cases, and only proceed if it lines up with the value of `s.unstable`.